ddprof-lib/src/main/cpp/javaApi.cpp

```diff
@@ -214,38 +224,66 @@ Java_com_datadoghq_profiler_JavaProfiler_recordSettingEvent0(JNIEnv *env, jobjec
     Profiler::instance()->writeDatadogProfilerSetting(tid, length, name_str.c_str(), value_str.c_str(), unit_str.c_str());
 }
 
+typedef struct {
+  u64 ticks;
+  u32 tid;
+} Origin;
+
+extern "C" DLLEXPORT jlong JNICALL
+Java_com_datadoghq_profiler_events_Origin_createTimer0(JNIEnv* env, jobject invalid, jint type) {
+    Origin* event = new Origin();
+    return (jlong) event;
```

Comment:
Could you add a counter to track created vs. freed ?
It would help us to spot any potential memory leak.